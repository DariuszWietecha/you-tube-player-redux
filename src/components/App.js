import React from 'react'
import Search from '../containers/Search'
import VisibleVideoList from '../containers/VisibleVideoList'
const App = () => (
  <div>
    <Search></Search>
    <VisibleVideoList></VisibleVideoList>
  </div>
)

export default App
