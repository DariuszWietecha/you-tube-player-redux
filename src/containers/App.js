import 'bootstrap/dist/css/bootstrap.css'
import { Container, Jumbotron } from 'reactstrap'
import React, { Component } from 'react'
import { connect } from 'react-redux'
import { fetchQuery, queryChange } from '../actions'
import VisibleNavigation from './VisibleNavigation'
import Search from './Search'
import Video from './VisibleVideo'
import VisibleVideoList from './VisibleVideoList'

class App extends Component {
  componentDidMount() {
    if (typeof this.props.match.params.query !== "undefined") {
      this.props.dispatch(queryChange(this.props.match.params.query))
      this.props.dispatch(fetchQuery(this.props.match.params.query))
    }
  }

  render() {
    return (
      <Container>
        <Jumbotron className="mt-4">
          <div className="display-4 mb-4 text-center">
            You Tube Player rcmodelreviews
          </div>
          <VisibleNavigation></VisibleNavigation>
          <Search></Search>
          <Video history={this.props.history}></Video>
          <VisibleVideoList></VisibleVideoList>
        </Jumbotron>
      </Container >
    )
  }
}

const mapDispatchToProps = dispatch => ({
  fetchQuery: (query) => dispatch(fetchQuery(query)),
  queryChange: (query) => dispatch(queryChange(query)),
})

export default connect(mapDispatchToProps)(App)

