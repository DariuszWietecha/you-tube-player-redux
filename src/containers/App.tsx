import "bootstrap/dist/css/bootstrap.css";
import React, { Component } from "react";
import { connect } from "react-redux";
import { RouteComponentProps } from "react-router-dom";
import { Container, Jumbotron } from "reactstrap";
import { fetchQuery, queryChange, resetSetUrl } from "../actions";
import * as actionsTypes from "../actions/types";
import * as commonTypes from "../commonTypes";
import {DivWithPointer, Overlay} from "../components/SimpleStyledComponents";
import Search from "./Search";
import VisibleNavigation from "./VisibleNavigation";
import Video from "./VisibleVideo";
import VisibleVideoList from "./VisibleVideoList";

interface IMatchParams {
  query: string;
}

interface IAppProps extends RouteComponentProps<IMatchParams> {
  isFetching: boolean;
  fetchQuery: (query: string) => commonTypes.DispatchAction;
  queryChange: (query: string) => actionsTypes.IQueryChangeAction;
  resetSetUrl: () => actionsTypes.IQueryChangeAction;
}

class App extends Component<IAppProps, {}> {
  public componentDidMount() {
    if (typeof this.props.match.params.query !== "undefined") {
      this.props.fetchQuery(this.props.match.params.query);
      this.props.queryChange(this.props.match.params.query);
    }
  }

  public render() {
    return (
      <div>
        <Overlay isFetching={this.props.isFetching}>
          <div>
            <div className="d-flex justify-content-center">
              <div className="spinner-border" role="status">
                <span className="sr-only">Loading...</span>
              </div>
            </div>
          </div>
        </Overlay>
        <Container className="pt-4">
          <Jumbotron>
            <DivWithPointer className="display-4 mb-4 text-center" onClick={this.props.resetSetUrl} role="button">
              You Tube Player
            </DivWithPointer>
            <VisibleNavigation></VisibleNavigation>
            <Search></Search>
            <Video></Video>
            <VisibleVideoList></VisibleVideoList>
          </Jumbotron>
        </Container >
      </div>
    );
  }
}

const mapStateToProps = (state: commonTypes.IApplicationState) => ({
  isFetching: state.searchResults.isFetching,
});

const mapDispatchToProps = (dispatch: any) => ({
  fetchQuery: (query: string) => dispatch(fetchQuery(query)),
  queryChange: (query: string) => dispatch(queryChange(query)),
  resetSetUrl: () => dispatch(resetSetUrl()),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
