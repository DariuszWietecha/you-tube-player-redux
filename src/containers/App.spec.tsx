import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import App from "./App";
import { routerTestProps } from "./routerTestProps";

describe("App container", () => {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const store = mockStore(
    {
      query: "query",
      searchResults: {
        index: 0,
        isFetching: false,
        items: [],
      },
    },
  );
  const { history, location, match } = routerTestProps("/:query?", { query: "query" });

  const props = {
    fetchQuery: jest.fn(),
    history,
    isFetching: false,
    location,
    match,
    queryChange: jest.fn(),
    resetSetUrl: jest.fn(),
  };

  it("Start view", () => {
    const component = renderer.create(
      <Provider store={store}>
        <App {...props} />
      </Provider>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
