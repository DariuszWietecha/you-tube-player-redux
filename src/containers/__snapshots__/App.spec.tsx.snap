// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container Start view 1`] = `
<div>
  <div
    className="sc-bdVaJa eXjbMa"
  >
    <div>
      <div
        className="d-flex justify-content-center"
      >
        <div
          className="spinner-border"
          role="status"
        >
          <span
            className="sr-only"
          >
            Loading...
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="pt-4 container"
  >
    <div
      className="jumbotron"
    >
      <div
        className="sc-bwzfXH npxoJ display-4 mb-4 text-center"
        onClick={[Function]}
        role="button"
      >
        You Tube Player
      </div>
      <form
        className=""
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <div
            className="input-group"
          >
            <input
              aria-describedby="button-addon4"
              autoComplete="off"
              autoFocus={true}
              className="form-control"
              list="searchHistory"
              name="query"
              onChange={[Function]}
              placeholder="Search in YouTube"
              type="search"
              value="query"
            />
            <datalist
              id="searchHistory"
            />
            <div
              className="input-group-append"
              id="button-addon4"
            >
              <button
                aria-label={null}
                className="btn btn-outline-primary"
                onClick={[Function]}
                type="button"
              >
                Search
              </button>
              <button
                aria-label={null}
                className="btn btn-outline-secondary"
                onClick={[Function]}
                type="button"
              >
                Reset
              </button>
            </div>
          </div>
        </div>
      </form>
      <div>
        <div
          className="row justify-content-end text-right"
        >
          <div
            className="d-none"
          >
            <div
              className="custom-control custom-switch"
            >
              <input
                checked={true}
                className="custom-control-input"
                id="customSwitch1"
                name="horizontal"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor="customSwitch1"
              >
                Switch to 
                grid
              </label>
            </div>
          </div>
        </div>
        <div
          className="d-flex flex-wrap"
        />
        <div
          className="d-none"
        >
          <button
            aria-label={null}
            className="btn btn-outline-secondary"
            onClick={[Function]}
            type="button"
          >
            More
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App container video item not set -> render null 1`] = `
<div>
  <div
    className="sc-bdVaJa eXjbMa"
  >
    <div>
      <div
        className="d-flex justify-content-center"
      >
        <div
          className="spinner-border"
          role="status"
        >
          <span
            className="sr-only"
          >
            Loading...
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="pt-4 container"
  >
    <div
      className="jumbotron"
    >
      <div
        className="sc-bwzfXH npxoJ display-4 mb-4 text-center"
        onClick={[Function]}
        role="button"
      >
        You Tube Player
      </div>
      <form
        className=""
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <div
            className="input-group"
          >
            <input
              aria-describedby="button-addon4"
              autoComplete="off"
              autoFocus={true}
              className="form-control"
              list="searchHistory"
              name="query"
              onChange={[Function]}
              placeholder="Search in YouTube"
              type="search"
              value="query"
            />
            <datalist
              id="searchHistory"
            />
            <div
              className="input-group-append"
              id="button-addon4"
            >
              <button
                aria-label={null}
                className="btn btn-outline-primary"
                onClick={[Function]}
                type="button"
              >
                Search
              </button>
              <button
                aria-label={null}
                className="btn btn-outline-secondary"
                onClick={[Function]}
                type="button"
              >
                Reset
              </button>
            </div>
          </div>
        </div>
      </form>
      <div>
        <div
          className="row justify-content-end text-right"
        >
          <div
            className="d-none"
          >
            <div
              className="custom-control custom-switch"
            >
              <input
                checked={true}
                className="custom-control-input"
                id="customSwitch1"
                name="horizontal"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="custom-control-label"
                htmlFor="customSwitch1"
              >
                Switch to 
                grid
              </label>
            </div>
          </div>
        </div>
        <div
          className="d-flex flex-wrap"
        />
        <div
          className="d-none"
        >
          <button
            aria-label={null}
            className="btn btn-outline-secondary"
            onClick={[Function]}
            type="button"
          >
            More
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
